schema.graphql

```diff
@@ -1,50 +1,76 @@
+# Use @goField(forceResolver: true) to lazily handle recursive or expensive fields that shouldn't be
+# resolved unless the caller asks for them
+directive @goField(
+  forceResolver: Boolean
+  name: String
+) on INPUT_FIELD_DEFINITION | FIELD_DEFINITION
+
+# Add @authRequired to any field that requires a user to be logged in. NOTE: Any field tagged with
+# @authRequired MUST return a union type that includes ErrNotAuthorized.
+#
+# If we need more control over auth in the future, this directive can be updated to take
+# arguments that specify the level of access required.
+directive @authRequired on FIELD_DEFINITION | INPUT_FIELD_DEFINITION
+
 scalar DateTime
 
 interface Node {
   id: ID!
 }
 
+interface Error {
+  message: String!
+}
+
 type GalleryUser implements Node {
   id: ID!
   username: String
   bio: String
   wallets: [Wallet]
-  gallery: Gallery
+  galleries: [Gallery] @goField(forceResolver: true)
```

Comment:
`@goField` here is actually used by gqlgen on the backend. The `forceResolver` argument enables lazy loading by telling gqlgen to generate a separate resolver for fetching that field's data. That separate resolver will only be invoked if the incoming query actually asks for the `galleries` field, so if you don't ask for it, the backend won't hit the database for it.

It's also really helpful for setting up circular references, since we do have cases where two types reference each other. E.g. a `GalleryUser` has `galleries: [Gallery]` and each `Gallery` has an `owner: GalleryUser`, so a naive implementation could infinitely recurse while trying to resolve those objects. A more clever (and more complex) implementation might cache references and set up the circle appropriately, but `forceResolver` obviates the need for any of that because you only set up the next reference if the query asks for it, so the recursion is limited to whatever is requested.

For example:
```
galleries {
  owner {
    galleries
      owner {
        username
      }
    }
  }
}
```

will do exactly what you asked it to do, recursing through gallery -> owner -> gallery -> owner, but no further.

Reply:
🔥 🔥 🔥

Reply:
that makes sense, thanks for the thorough explanation @radazen !